Graphs/BinaryLifting.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Author: Adrito Mukherjee
+ * Binary Lifting implementation in Javascript
+ * Binary Lifting is a technique that is used to find the kth ancestor of a node in a rooted tree with N nodes
+ * The technique requires preprocessing the tree in O(N log N) using dynamic programming
+ * The techniqe can answer Q queries about kth ancestor of any node in O(Q log N)
+ * It is faster than the naive algorithm that answers Q queries with complexity O(Q K)
+ * It can be used to find Lowest Common Ancestor of two nodes in O(log N)
+ * Tutorial on Binary Lifting: https://codeforces.com/blog/entry/100826
+ */
+
+class BinaryLifting {
+  constructor (root, tree) {
+    this.connections = {}
+    this.up = {} // up[node][i] stores the 2^i-th  parent of node
+    for (const [i, j] of tree) {
+      this.addEdge(i, j)
+    }
+    // LOG should be such that 2^LOG is greater than total number of nodes in the tree
+    this.LOG = 0
```

Comment:
This name violates our naming conventions. Please name it `log` (lowercase).

Reply:
changed `LOG` to `log`